go/mysql/binlog_event_json.go

```diff
@@ -36,23 +54,49 @@ func jlog(tpl string, vals ...interface{}) {
 	if !jsonDebug {
 		return
 	}
-	fmt.Printf(tpl+"\n", vals...)
+	log.Infof("JSON:"+tpl+"\n", vals...)
 	_ = printASCIIBytes
 }
 
 func printASCIIBytes(data []byte) {
 	if !jsonDebug {
 		return
 	}
-	fmt.Printf("\n\n%v\n[", data)
+	s := ""
 	for _, c := range data {
 		if c < 127 && c > 32 {
-			fmt.Printf("%c ", c)
+			s += fmt.Sprintf("%c ", c)
 		} else {
-			fmt.Printf("%02d ", c)
+			s += fmt.Sprintf("%02d ", c)
 		}
 	}
-	fmt.Printf("]\n")
+	log.Infof("[%s]", s)
+}
+
+// only used for logging/debugging
+var jsonTypeToName = map[uint]string{
```

Comment:
is there no `boolean` type?

Reply:
the `literal` type is used for booleans and nulls